### PR TITLE
[Security] Fix for #SEC-50 [#OSF-6972]

### PR DESCRIPTION
**Pull request by cslzchen**
## Ticket
https://openscience.atlassian.net/browse/OSF-6972
https://openscience.atlassian.net/browse/SEC-50

## Purpose

Fix security issue described in https://openscience.atlassian.net/browse/SEC-50. Please refer to this ticket for detailed information.

## Side effects

No

### CODE REVIEW

File: framework/auth/views.py [hunk 1]
```diff
@@ -245,13 +245,8 @@ def auth_login(auth, **kwargs):
...
 
-    if next_url:
-        # Only allow redirects which are relative root or full domain, disallows external redirects.
-        if not (next_url[0] == '/'
-                or next_url.startswith(settings.DOMAIN)
-                or next_url.startswith(settings.CAS_SERVER_URL)
-                or next_url.startswith(settings.MFR_SERVER_URL)):
-            raise HTTPError(http.InvalidURL)
+    if not validate_next_url(next_url):
+        raise HTTPError(http.BAD_REQUEST)
```
Comment (cslzchen) [inline framework/auth/views.py]: For Reviewer: OSF cannot handle `http.InvalidURL`, change to `http.BAD_REQUEST` which is more appropriate in this case.

File: framework/auth/views.py [hunk 2]
```diff
@@ -825,3 +820,27 @@ def external_login_email_post():
         'form': form,
         'external_id_provider': external_id_provider
     }
+
+
+def validate_next_url(next_url):
```
Comment (cslzchen) [inline framework/auth/views.py]: For Reviewer: this is used in the another PR https://github.com/CenterForOpenScience/osf.io/pull/6082 (CAS Institution Login): https://github.com/CenterForOpenScience/osf.io/pull/6082/files#diff-9a64b4982d0bd80f7a498c65ddce2023R283.

**Comment by brianjgeiger** [comment]
Can you make this a hotfix?

**Comment by cslzchen** [comment]
Sure. I will close this PR and make it a hot fix.

**Pull request closed without merging**